orchestrator/test_runner/src/happy_path_v2.rs

```diff
@@ -170,27 +177,37 @@ pub async fn happy_path_test_v2(
     info!("Sent batch request to move things along");
 
     info!("Waiting for batch to be signed and relayed to Ethereum");
-    let start = Instant::now();
-    while Instant::now() - start < TOTAL_TIMEOUT {
-        let balance = web30
+
+    match tokio::time::timeout(TOTAL_TIMEOUT, async {
+        loop {
+            let balance = web30
             .get_erc20_balance(erc20_contract, user.eth_address)
             .await;
         if balance.is_err() {
             continue;
         }
         let balance = balance.unwrap();
         if balance == amount_to_bridge {
-            info!(
-                "Successfully bridged {} Cosmos asset {} to Ethereum!",
-                amount_to_bridge, token_to_send_to_eth
-            );
             break;
         } else if balance != 0u8.into() {
             panic!(
                 "Expected {} {} but got {} instead",
                 amount_to_bridge, token_to_send_to_eth, balance
             );
         }
-        delay_for(Duration::from_secs(1)).await;
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+        }
+    }).await {
+        Ok(_) => {
+            info!(
+                "Successfully bridged {} Cosmos asset {} to Ethereum!",
+                amount_to_bridge, token_to_send_to_eth
+            );
+        },
+        Err(_) => {
+            panic!(
+                "An error occured",
```

Comment:
Maybe a little more description of the error?

Reply:
This error fixes this:  https://github.com/PeggyJV/gravity-bridge/pull/274/commits/81c152c69f81e4bdd89234789794dcd0b46f3730

An early panic will also be thrown if balance don't match amount to be bridged.

```rust
                panic!(
                    "Expected {} {} but got {} instead",
                    amount_to_bridge, token_to_send_to_eth, balance
                );
```